tests/ui/crashes/ice-8850.fixed

```diff
@@ -1,15 +1,15 @@
 fn fn_pointer_static() -> usize {
     static FN: fn() -> usize = || 1;
     
-    FN() + 1
+    (FN() + 1)
```

Comment:
using `Sugg::maybe_par` seems like a easy fix, but the drawback is these unncessary parentheses, should this be fine or should it add parentheses only if the binary ops are consider... "complex"? (meaning have branches or something like that)

Reply:
The `needless_return` lint also has some logic for rewriting `return if true { 1 } else { 2 } + 3;` to `(if true { 1 } else { 2 } + 3)`.

It would be nice if we can reuse that function here. https://github.com/rust-lang/rust-clippy/blob/master/clippy_lints/src/returns.rs#L407
Might be worth moving to utils?